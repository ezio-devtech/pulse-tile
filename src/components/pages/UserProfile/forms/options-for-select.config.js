const optionsForThemesField = [{
  value: 'green',
  title: 'Green Theme',
},
{
  value: 'purple',
  title: 'Purple Theme',
},
{
  value: 'gray',
  title: 'Gray Theme',
}];

const optionsForGenderField = [{
  value: 'female',
  title: 'Female',
}, {
  value: 'male',
  title: 'Male',
}];

export { optionsForThemesField, optionsForGenderField }
