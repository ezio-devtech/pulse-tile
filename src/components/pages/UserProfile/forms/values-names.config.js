export const valuesNames = {
  APP_TITLE: 'title',
  LOGO_PATH: 'logoPath',
  SELECT_THEME: 'selectThemeField',
  BROWSER_TITLE: 'browserWindowTitle',
};
