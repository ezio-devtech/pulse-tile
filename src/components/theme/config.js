import vaccinations from './plugins/Vaccinations/index';

/**
 * The list of none-core plugins which are included to the current theme
 *
 * @return array
 */
export const themePlugins = [

    vaccinations,

];

/**
 * The list of clients URLs for none-core plugins
 *
 * @return array
 */
export const themeClientUrls = {

    VACCINATIONS: 'vaccinations',

};

/**
 * The list of pages for none-core plugins
 *
 * @return array
 */
export const themePluginsPages = {

    'vaccinations': {
      breadcrumbs: [{
        title: 'Vaccinations',
        state: '/vaccinations',
      }],
    },

};

/**
 *
 */
export const themePatientSummaryConfig = [
    {
        key: 'vaccinations',
        title: 'Vaccinations',
        titleCheckboxes: 'Vaccinations',
        state: 'vaccinations',
        nameCheckboxes: 'vaccinations',
        imgPreview: '/images/patients-summary/vaccinations.jpg',
        isDefaultSelected: true,
    },
];
