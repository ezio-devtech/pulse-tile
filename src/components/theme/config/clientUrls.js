/**
 * This information is necessary for plugins URLs
 */
export const themeClientUrls = {
    PATIENTS: '/patients',
};
