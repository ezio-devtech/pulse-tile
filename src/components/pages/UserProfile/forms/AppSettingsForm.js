import React, { PureComponent } from 'react';
import { Field, reduxForm } from 'redux-form'
import { Row, Col } from 'react-bootstrap';

import ValidatedInput from '../../../form-fields/ValidatedInputFormGroup';
import FileInput from '../../../form-fields/FileInput';
import { optionsForThemesField } from './options-for-select.config';
import Select from '../../../form-fields/SelectFormGroup';

@reduxForm({ form: 'appSettingsFormSelector' })
export default class AppSettingsForm extends PureComponent {
  render() {
    return (
      <div className="panel-body-inner">
        <form name="appSettingsForm" className="form">
          <div className="form-group-wrapper">
            <Row>
              <Col xs={12} md={6}>
                <Row>
                  <Col md={11}>
                    <Field
                      label="Application Title"
                      name="applicationTitle"
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label="Application Logo File"
                      name="logoPath"
                      component={FileInput}
                      id="logoPath"
                    />
                    <Field
                      label="Application Themes"
                      name="green"
                      component={Select}
                      options={optionsForThemesField}
                    />
                    <Field
                      label="Browser Window title"
                      name="browserWindowTitle"
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                  </Col>
                </Row>
              </Col>
            </Row>
          </div>
        </form>
      </div>)
  }
}
