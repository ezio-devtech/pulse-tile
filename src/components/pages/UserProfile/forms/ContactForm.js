import React, { PureComponent } from 'react';
import { Field, reduxForm } from 'redux-form'
import { Row, Col } from 'react-bootstrap';

import ValidatedInput from '../../../form-fields/ValidatedInputFormGroup';
import { optionsForCountryField } from './options-for-select.config';
import Select from '../../../form-fields/SelectFormGroup';
import { validateContactForm } from './validation';

@reduxForm({
  form: 'contactFormSelector',
  validate: validateContactForm,
})
export default class ContactForm extends PureComponent {
  render() {
    return (
      <div className="panel-body-inner">
        <form name="personalForm" className="form">
          <div className="form-group-wrapper">
            <Row>
              <Col xs={12} md={6}>
                <Row>
                  <Col md={11}>
                    <Field
                      label="Address"
                      name="address"
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label="City"
                      name="city"
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label="State"
                      name="state"
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label="Postal Code"
                      name="postalCode"
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label="Country"
                      name="country"
                      placeholder=""
                      component={Select}
                      options={optionsForCountryField}
                    />
                  </Col>
                </Row>
              </Col>
              <Col xs={12} md={6}>
                <Row>
                  <Col md={11} mdOffset={1}>
                    <Field
                      label="Phone Number"
                      name="phone"
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label="Email"
                      name="email"
                      placeholder=""
                      component={ValidatedInput}
                    />
                  </Col>
                </Row>
              </Col>
            </Row>
          </div>
        </form>
      </div>)
  }
}
