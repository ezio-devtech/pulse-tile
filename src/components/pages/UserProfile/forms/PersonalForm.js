import React, { PureComponent } from 'react';
import { Field, reduxForm } from 'redux-form'
import { Row, Col } from 'react-bootstrap';

import ValidatedInput from '../../../form-fields/ValidatedInputFormGroup';
import DateInput from '../../../form-fields/DateInput';
import { optionsForGenderField } from './options-for-select.config';
import Select from '../../../form-fields/SelectFormGroup';
import { validatePersonalForm } from './validation';

@reduxForm({
  form: 'personalFormSelector',
  validate: validatePersonalForm,
})
export default class PersonalForm extends PureComponent {
  render() {
    return (
      <div className="panel-body-inner">
        <form name="personalForm" className="form">
          <div className="form-group-wrapper">
            <Row>
              <Col xs={12} md={6}>
                <Row>
                  <Col md={11}>
                    <Field
                      label="First Name"
                      name="firstname"
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label="Last Name"
                      name="lastname"
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label="NHS No"
                      name="nhs"
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                  </Col>
                </Row>
              </Col>
              <Col xs={12} md={6}>
                <Row>
                  <Col md={11} mdOffset={1}>
                    <Field
                      label="Date of Birth"
                      name="birthday"
                      type="text"
                      placeholder=""
                      component={DateInput}
                    />
                    <Field
                      label="Gender"
                      name="gender"
                      placeholder=""
                      component={Select}
                      options={optionsForGenderField}
                    />
                    <Field
                      label="Doctor"
                      name="doctor"
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                  </Col>
                </Row>
              </Col>
            </Row>
          </div>
        </form>
      </div>)
  }
}
