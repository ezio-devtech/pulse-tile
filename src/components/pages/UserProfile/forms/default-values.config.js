import { valuesSettingsForm, valuesPersonalForm, valuesContactForm } from './values-names.config';

const defaultPersonalFormValues = {
  [valuesPersonalForm.FIRST_NAME]: 'Bob',
  [valuesPersonalForm.LAST_NAME]: 'Smith',
  [valuesPersonalForm.SELECT_GENDER]: 'female',
  [valuesPersonalForm.DOCTOR]: 'Dr Emma Huston',
};

const defaultContactFormValues = {
  [valuesContactForm.ADDRESS]: '6801 Tellus Street',
  [valuesContactForm.CITY]: 'Westmorland',
  [valuesContactForm.STATE]: 'Westmorland',
  [valuesContactForm.POSTAL_CODE]: 'Box 306',
  [valuesContactForm.PHONE]: '07624 647524',
  [valuesContactForm.EMAIL]: 'bob.smith@gmail.com',
};

const defaultAppSettingsFormValues = {
  [valuesSettingsForm.APP_TITLE]: 'should be fetched from server',
  [valuesSettingsForm.LOGO_PATH]: 'should be fetched from server',
  [valuesSettingsForm.SELECT_THEME]: 'themeDefault',
  [valuesSettingsForm.BROWSER_TITLE]: 'should be fetched from server',
};

export { defaultPersonalFormValues, defaultContactFormValues, defaultAppSettingsFormValues }
