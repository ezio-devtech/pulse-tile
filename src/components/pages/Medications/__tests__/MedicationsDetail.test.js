import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import { get } from 'lodash';

import MedicationsDetail from '../MedicationsDetail/MedicationsDetail';
import { valuesNames, valuesLabels } from '../forms.config';
import { getDDMMMYYYY } from '../../../../utils/time-helpers.utils';
import { themeConfigs } from '../../../../themes.config';
import { isShowElement, isPanelVisible } from '../../../../utils/themeSettings-helper';

Enzyme.configure({ adapter: new Adapter() });

/**
 * This function calculate number of details panel which should be shown
 *
 * @return {number}
 */
function getDetailPanelsNumber(hiddenPanels) {
  const maximalTotalNumber = 4;
  return maximalTotalNumber - hiddenPanels.length;
}

const hiddenPanels = get(themeConfigs, 'panelsToHide.medications', []);

const propsForMedicationsPanel = {
  detail: {
    [valuesNames.NAME]: 'test',
    [valuesNames.DOSE_AMOUNT]: '50',
    [valuesNames.DOSE_DIRECTIONS]: 'test',
    [valuesNames.DOSE_TIMING]: '50',
    [valuesNames.ROUTE]: 'test',
    [valuesNames.START_DATE]: 1510653645000,
    [valuesNames.START_TIME]: 36045000,
    [valuesNames.MEDICATION_CODE]: 123456789,
    [valuesNames.MEDICATION_TERMINOLOGY]: 'external',
    [valuesNames.AUTHOR]: 'bob.smith@gmail.com',
    [valuesNames.DATE_CREATED]: 1510588832000,
    [valuesNames.SOURCE]: 'marand',
    [valuesNames.SOURCE_ID]: 'a7007401-837f-471c-8f73-cbb53c0eb1a1',
  },
};

const onShow = () => {};
const toggleHourlySchedule = () => {};

const MEDICATION_PANEL = 'medicationPanel';
const PRESCRIPTION_PANEL = 'prescriptionPanel';
const WARNINGS_PANEL = 'warningsPanel';
const CHANGE_HISTORY_PANEL = 'changeHistoryPanel';

const CONVERT_DATE_CREATED = getDDMMMYYYY(propsForMedicationsPanel.detail[valuesNames.DATE_CREATED]);

const hideElements = get(themeConfigs, 'detailsToHide.medications', []);

describe('Component <MedicationsDetail />', () => {
  it('should renders with props correctly', () => {
    const component = shallow(<MedicationsDetail
      onShow={onShow}
      toggleHourlySchedule={toggleHourlySchedule}
    />);
    //
    // Testing component when detail filled object, expandedPanel is all, and panel not edited
    component.setProps({ detail: propsForMedicationsPanel.detail, expandedPanel: 'all', editedPanel: { [MEDICATION_PANEL]: false } });
    expect(component.props().className).toEqual('section-detail');

    const panelsNumber = getDetailPanelsNumber(hiddenPanels);

    expect(component.find('PluginDetailPanel')).toHaveLength(panelsNumber);
    expect(component.find('MedicationsDetailPanel')).toHaveLength(1);

    // Testing medicationPanel
    expect(component.find('MedicationsDetailPanel').props().name).toEqual(MEDICATION_PANEL);
    expect(component.find('MedicationsDetailPanel').props().title).toEqual('Medication');
    expect(component.find('MedicationsDetailPanel').props().isOpen).toEqual(false);
    expect(component.find('MedicationsDetailPanel').props().isBtnShowPanel).toEqual(true);
    expect(component.find('MedicationsDetailPanel').props().isShowControlPanel).toEqual(true);

    let count = 0;

    if (isShowElement(valuesNames.NAME, hideElements)) {
      expect(component.find('MedicationsDetailPanel').find('.control-label').at(count).text()).toEqual(valuesLabels.NAME);
      expect(component.find('MedicationsDetailPanel').find('.form-control-static').at(count).text()).toEqual(propsForMedicationsPanel.detail[valuesNames.NAME]);
      count++;
    }

    if (isShowElement(valuesNames.DOSE_AMOUNT, hideElements)) {
      expect(component.find('MedicationsDetailPanel').find('.control-label').at(count).text()).toEqual(valuesLabels.DOSE_AMOUNT);
      expect(component.find('MedicationsDetailPanel').find('.form-control-static').at(count).text()).toEqual(propsForMedicationsPanel.detail[valuesNames.DOSE_AMOUNT]);
      count++;
    }

    if (isShowElement(valuesNames.DOSE_TIMING, hideElements)) {
      expect(component.find('MedicationsDetailPanel').find('.control-label').at(count).text()).toEqual(valuesLabels.DOSE_TIMING);
      expect(component.find('MedicationsDetailPanel').find('.form-control-static').at(count).text()).toEqual(propsForMedicationsPanel.detail[valuesNames.DOSE_TIMING]);
      component.find('.btn-primary').simulate('click');
      count++;
    }

    if (isShowElement(valuesNames.DOSE_DIRECTIONS, hideElements)) {
      expect(component.find('MedicationsDetailPanel').find('.control-label').at(count).text()).toEqual(valuesLabels.DOSE_DIRECTIONS);
      expect(component.find('MedicationsDetailPanel').find('.form-control-static').at(count).text()).toEqual(propsForMedicationsPanel.detail[valuesNames.DOSE_DIRECTIONS]);
      count++;
    }

    if (isShowElement(valuesNames.AUTHOR, hideElements)) {
      expect(component.find('MedicationsDetailPanel').find('.control-label').at(count).text()).toEqual(valuesLabels.AUTHOR);
      expect(component.find('MedicationsDetailPanel').find('.form-control-static').at(count).text()).toEqual(propsForMedicationsPanel.detail[valuesNames.AUTHOR]);
      count++;
    }

    if (isShowElement(valuesNames.DATE_CREATED, hideElements)) {
      expect(component.find('MedicationsDetailPanel').find('.control-label').at(count).text()).toEqual(valuesLabels.DATE_CREATED);
      expect(component.find('MedicationsDetailPanel').find('.form-control-static').at(count).text()).toEqual(CONVERT_DATE_CREATED);
      count++;
    }

    // Testing prescriptionPanel
    let countPanels = 0;
    if (isPanelVisible(hiddenPanels, 'prescription')) {
        expect(component.find('PluginDetailPanel').at(countPanels).props().name).toEqual(PRESCRIPTION_PANEL);
        expect(component.find('PluginDetailPanel').at(countPanels).props().title).toEqual('Prescription (1)');
        expect(component.find('PluginDetailPanel').at(countPanels).props().isOpen).toEqual(false);
        expect(component.find('PluginDetailPanel').at(countPanels).props().isBtnShowPanel).toEqual(true);
        expect(component.find('PluginDetailPanel').at(countPanels).props().isShowControlPanel).toEqual(true);
        expect(component.find('PluginDetailPanel').at(countPanels).find('.control-label').at(0).text()).toEqual(valuesLabels.NAME);
        expect(component.find('PluginDetailPanel').at(countPanels).find('.control-label').at(1).text()).toEqual(valuesLabels.DOSE_AMOUNT);
        expect(component.find('PluginDetailPanel').at(countPanels).find('.control-label').at(2).text()).toEqual(valuesLabels.DOSE_TIMING);
        expect(component.find('PluginDetailPanel').at(countPanels).find('.form-control-static').at(0).text()).toEqual(propsForMedicationsPanel.detail[valuesNames.NAME]);
        expect(component.find('PluginDetailPanel').at(countPanels).find('.form-control-static').at(1).text()).toEqual(propsForMedicationsPanel.detail[valuesNames.DOSE_AMOUNT]);
        expect(component.find('PluginDetailPanel').at(countPanels).find('.form-control-static').at(2).text()).toEqual('2X each morning');

        component.setProps({ isOpenHourlySchedule: true });
        expect(component.find('.panel-title').text()).toEqual('27-Nov-2016');

        countPanels++;
    }

    // Testing warningsPanel
    if (isPanelVisible(hiddenPanels, 'warnings')) {
      expect(component.find('PluginDetailPanel').at(countPanels).props().name).toEqual(WARNINGS_PANEL);
      expect(component.find('PluginDetailPanel').at(countPanels).props().title).toEqual('Warnings (2)');
      expect(component.find('PluginDetailPanel').at(countPanels).props().isOpen).toEqual(false);
      expect(component.find('PluginDetailPanel').at(countPanels).props().isBtnShowPanel).toEqual(true);
      expect(component.find('PluginDetailPanel').at(countPanels).props().isShowControlPanel).toEqual(false);
      expect(component.find('PluginDetailPanel').at(countPanels).find('.control-label').at(0).text()).toEqual('Warning #1');
      expect(component.find('PluginDetailPanel').at(countPanels).find('.control-label').at(1).text()).toEqual('Effects');
      expect(component.find('PluginDetailPanel').at(countPanels).find('.control-label').at(2).text()).toEqual('Warning #2');
      expect(component.find('PluginDetailPanel').at(countPanels).find('.control-label').at(3).text()).toEqual('Effects');
      expect(component.find('PluginDetailPanel').at(countPanels).find('.form-control-static').at(0).text()).toEqual('Interaction found between Furosemide and Latanoprost');
      expect(component.find('PluginDetailPanel').at(countPanels).find('.form-control-static').at(1).text()).toEqual('Anticonvulsant effect antagonised');
      expect(component.find('PluginDetailPanel').at(countPanels).find('.form-control-static').at(2).text()).toEqual('Interaction found between Furosemide and Digoxin');
      expect(component.find('PluginDetailPanel').at(countPanels).find('.form-control-static').at(3).text()).toEqual('May increase anticoagulant effect');
      countPanels++;
    }

    // Testing CHANGE_HISTORY_PANEL
    if (isPanelVisible(hiddenPanels, 'history')) {
      expect(component.find('PluginDetailPanel').at(countPanels).props().name).toEqual(CHANGE_HISTORY_PANEL);
      expect(component.find('PluginDetailPanel').at(countPanels).props().title).toEqual('Change History (1)');
      expect(component.find('PluginDetailPanel').at(countPanels).props().isOpen).toEqual(false);
      expect(component.find('PluginDetailPanel').at(countPanels).props().isBtnShowPanel).toEqual(true);
      expect(component.find('PluginDetailPanel').at(countPanels).props().isShowControlPanel).toEqual(false);
      expect(component.find('PluginDetailPanel').at(countPanels).find('.control-label').at(0).text()).toEqual('Change #1 Date');
      expect(component.find('PluginDetailPanel').at(countPanels).find('.control-label').at(1).text()).toEqual('Changes');
      expect(component.find('PluginDetailPanel').at(countPanels).find('.form-control-static').at(0).text()).toEqual('11-Oct-2016 11:45');

      component.find('.btn-schedule').simulate('click');
      expect(component).toMatchSnapshot();

      countPanels++;
    }

    component.setProps({ detail: { [valuesNames.ISIMPORT]: true } });
    // expect(component.find('.form-control-static').at(4).text()).toEqual('');
    // expect(component.find('.control-label').at(4).text()).toEqual(valuesLabels.ORIGINAL_SOURCE);
    // expect(component.find('.control-label').at(5).text()).toEqual(valuesLabels.ISIMPORT);
    expect(component.find('Switch')).toHaveLength(1);
  });

  it('should renders correctly with different state of props', () => {
    const component = shallow(
      <MedicationsDetail />);
    // Testing component when detail empty object, expandedPanel is medicationPanel
    component.setProps({ detail: { [valuesNames.DATE_CREATED]: 1510588832000 }, expandedPanel: MEDICATION_PANEL, editedPanel: { [MEDICATION_PANEL]: false } });
    expect(component.find('PluginDetailPanel')).toHaveLength(0);
    expect(component.find('MedicationsDetailPanel')).toHaveLength(1);
    expect(component.find('MedicationsDetailPanel').props().name).toEqual(MEDICATION_PANEL);
    expect(component).toMatchSnapshot();

    // Testing component when detail empty object, expandedPanel is prescriptionPanel
    if (isPanelVisible(hiddenPanels, 'prescription')) {
      component.setProps({ detail: { [valuesNames.DATE_CREATED]: 1510588832000 }, expandedPanel: PRESCRIPTION_PANEL });
      expect(component.find('PluginDetailPanel')).toHaveLength(1);
      expect(component.find('MedicationsDetailPanel')).toHaveLength(0);
      expect(component.find('PluginDetailPanel').props().name).toEqual(PRESCRIPTION_PANEL);
      expect(component).toMatchSnapshot();
    }

    // Testing component when detail empty object, expandedPanel is warningsPanel
    if (isPanelVisible(hiddenPanels, 'warnings')) {
      component.setProps({ detail: { [valuesNames.DATE_CREATED]: 1510588832000 }, expandedPanel: WARNINGS_PANEL });
      expect(component.find('PluginDetailPanel')).toHaveLength(1);
      expect(component.find('MedicationsDetailPanel')).toHaveLength(0);
      expect(component.find('PluginDetailPanel').props().name).toEqual(WARNINGS_PANEL);
      expect(component).toMatchSnapshot();
    }

    // Testing component when detail empty object, expandedPanel is changeHistoryPanel
    if (isPanelVisible(hiddenPanels, 'history')) {
      component.setProps({
        detail: {[valuesNames.DATE_CREATED]: 1510588832000},
        expandedPanel: CHANGE_HISTORY_PANEL
      });
      expect(component.find('PluginDetailPanel')).toHaveLength(1);
      expect(component.find('MedicationsDetailPanel')).toHaveLength(0);
      expect(component.find('PluginDetailPanel').props().name).toEqual(CHANGE_HISTORY_PANEL);
      expect(component).toMatchSnapshot();
    }

    // Testing component when detail filled object, expandedPanel is all, and MEDICATION_PANEL is edited
    component.setProps({ detail: propsForMedicationsPanel.detail, expandedPanel: MEDICATION_PANEL, editedPanel: { [MEDICATION_PANEL]: true } });
    expect(component.find('ReduxForm')).toHaveLength(1);
    expect(component.find('PluginDetailPanel').props().name).toEqual(MEDICATION_PANEL);
    expect(component).toMatchSnapshot();

    // Testing component when detail filled object, expandedPanel is all, and PRESCRIPTION_PANEL is edited
    component.setProps({ detail: propsForMedicationsPanel.detail, expandedPanel: PRESCRIPTION_PANEL, editedPanel: { [PRESCRIPTION_PANEL]: true } });
    expect(component.find('ReduxForm')).toHaveLength(1);
    expect(component.find('PluginDetailPanel').props().name).toEqual(PRESCRIPTION_PANEL);
    expect(component).toMatchSnapshot();
  });
});

