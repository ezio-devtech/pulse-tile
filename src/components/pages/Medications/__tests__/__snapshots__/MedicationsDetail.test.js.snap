// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <MedicationsDetail /> should renders correctly with different state of props 1`] = `
<div
  className="section-detail"
>
  <div
    className="panel-group accordion"
  >
    <MedicationsDetailPanel
      editedPanel={
        Object {
          "medicationPanel": false,
        }
      }
      isBtnShowPanel={true}
      isOpen={false}
      isShowControlPanel={true}
      name="medicationPanel"
      title="Medication"
    >
      <div
        className="panel-body-inner"
      >
        <div
          className="form"
        >
          <div
            className="form-group-wrapper"
          >
            <div
              className="row-expand"
            >
              <div
                className="col-expand-left"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    Name
                  </label>
                  <div
                    className="form-control-static"
                  />
                </div>
              </div>
            </div>
            <div
              className="row-expand"
            >
              <div
                className="col-expand-left"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    Dose Descriptions
                  </label>
                  <div
                    className="form-control-static"
                  />
                </div>
              </div>
            </div>
            <div
              className="row-expand"
            >
              <div
                className="col-expand-left"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    Author
                  </label>
                  <div
                    className="form-control-static"
                  />
                </div>
              </div>
            </div>
            <div
              className="row-expand"
            >
              <div
                className="col-expand-left"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    Date
                  </label>
                  <div
                    className="form-control-static"
                  >
                    13-Nov-2017
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </MedicationsDetailPanel>
  </div>
</div>
`;

exports[`Component <MedicationsDetail /> should renders correctly with different state of props 2`] = `
<div
  className="section-detail"
>
  <div
    className="panel-group accordion"
  >
    <PluginDetailPanel
      editedPanel={
        Object {
          "medicationPanel": true,
        }
      }
      isBtnShowPanel={true}
      isEditButton={true}
      isOpen={false}
      isShowControlPanel={true}
      name="medicationPanel"
      title="Edit Medication"
    >
      <ReduxForm
        detail={
          Object {
            "author": "bob.smith@gmail.com",
            "dateCreated": 1510588832000,
            "doseAmount": "50",
            "doseDirections": "test",
            "doseTiming": "50",
            "medicationCode": 123456789,
            "medicationTerminology": "external",
            "name": "test",
            "route": "test",
            "source": "marand",
            "sourceId": "a7007401-837f-471c-8f73-cbb53c0eb1a1",
            "startDate": 1510653645000,
            "startTime": 36045000,
          }
        }
      />
    </PluginDetailPanel>
  </div>
</div>
`;

exports[`Component <MedicationsDetail /> should renders correctly with different state of props 3`] = `
<div
  className="section-detail"
>
  <div
    className="panel-group accordion"
  >
    <PluginDetailPanel
      editedPanel={
        Object {
          "prescriptionPanel": true,
        }
      }
      isBtnShowPanel={true}
      isEditButton={true}
      isOpen={false}
      isShowControlPanel={true}
      name="prescriptionPanel"
      title="Edit Prescription (1)"
    >
      <ReduxForm
        detail={
          Object {
            "author": "bob.smith@gmail.com",
            "dateCreated": 1510588832000,
            "doseAmount": "50",
            "doseDirections": "test",
            "doseTiming": "50",
            "medicationCode": 123456789,
            "medicationTerminology": "external",
            "name": "test",
            "route": "test",
            "source": "marand",
            "sourceId": "a7007401-837f-471c-8f73-cbb53c0eb1a1",
            "startDate": 1510653645000,
            "startTime": 36045000,
          }
        }
      />
    </PluginDetailPanel>
  </div>
</div>
`;
